x/ibc/core/03-connection/types/msgs.go

```diff
@@ -116,15 +116,19 @@ func (msg MsgConnectionOpenTry) Type() string {
 
 // ValidateBasic implements sdk.Msg
 func (msg MsgConnectionOpenTry) ValidateBasic() error {
-	if err := host.ConnectionIdentifierValidator(msg.DesiredConnectionId); err != nil {
-		return sdkerrors.Wrap(err, "invalid desired connection ID")
-	}
-	if msg.CounterpartyChosenConnectionId != "" && msg.CounterpartyChosenConnectionId != msg.DesiredConnectionId {
-		return sdkerrors.Wrap(ErrInvalidConnectionIdentifier, "counterparty chosen connection identifier must be empty or equal to desired connection identifier")
+	// an empty connection identifier indicates that a connection identifier should be generated
+	if msg.PreviousConnectionId != "" {
+		if err := host.ConnectionIdentifierValidator(msg.PreviousConnectionId); err != nil {
```

Comment:
I believe we can be even stricter with our own connection id and make sure it is in the deterministic format we expect?

Reply:
didn't address this because it is a nice-to-have imo and I'm not sure how to do it quickly without spending time looking it up. It can be included in a followup if someone else wants to pick it up

Reply:
I ended up doing this but with string parsing. I needed the helper function for the escrow address pr so figured I might as well